RecoTracker/MkFit/plugins/createPhase1TrackerGeometry.cc

```diff
@@ -0,0 +1,372 @@
+        z_at_outer_brl = S.zAtR(outer_brl.m_rout);
+        if (z_dir_pos)
+          misses_first_tec = z_at_outer_brl < tec_first.m_zmin;
+        else
+          misses_first_tec = z_at_outer_brl > tec_first.m_zmax;
+      } else {
+        z_at_maxr = S.zAtR(maxR);
+        if (z_dir_pos)
+          misses_first_tec = z_at_maxr < tec_first.m_zmin;
+        else
+          misses_first_tec = z_at_maxr > tec_first.m_zmax;
+      }
+
+      if (/*can_reach_outer_brl &&*/ misses_first_tec)
+      // outer_brl.is_within_z_limits(S.zAtR(outer_brl.r_mean())))
+      {
+        reg = TrackerInfo::Reg_Barrel;
+      } else {
+        // This should be a list of layers
+        // CMS, first tib, tob: 4, 10
+
+        if ((S.canReachRadius(tib1.m_rin) && tib1.is_within_z_limits(S.zAtR(tib1.m_rin))) ||
+            (S.canReachRadius(tob1.m_rin) && tob1.is_within_z_limits(S.zAtR(tob1.m_rin)))) {
+          // transition region ... we are still hitting barrel layers
+
+          reg = z_dir_pos ? TrackerInfo::Reg_Transition_Pos : TrackerInfo::Reg_Transition_Neg;
+        } else {
+          // endcap ... no barrel layers will be hit anymore.
+
+          reg = z_dir_pos ? TrackerInfo::Reg_Endcap_Pos : TrackerInfo::Reg_Endcap_Neg;
+        }
+      }
+
+      part.m_region[i] = reg;
+      part.m_sort_score[i] = 5.0f * (reg - 2) + eta;
+    }
+  }
+}  // namespace
+
+namespace mkfit {
+  void createPhase1TrackerGeometry(TrackerInfo &ti, IterationsInfo &ii, bool verbose) {
+    // TODO: these writes to global variables need to be removed
+    Config::nTotalLayers = 18 + 2 * 27;
+
+    Config::useCMSGeom = true;
+
+    Config::finding_requires_propagation_to_hit_pos = true;
+    Config::finding_inter_layer_pflags = PropagationFlags(PF_use_param_b_field | PF_apply_material);
+    Config::finding_intra_layer_pflags = PropagationFlags(PF_none);
+    Config::backward_fit_pflags = PropagationFlags(PF_use_param_b_field | PF_apply_material);
+    Config::forward_fit_pflags = PropagationFlags(PF_use_param_b_field | PF_apply_material);
+    Config::seed_fit_pflags = PropagationFlags(PF_none);
+    Config::pca_prop_pflags = PropagationFlags(PF_none);
+
+    ti.set_eta_regions(0.9, 1.7, 2.45, false);
+    ti.create_layers(18, 27, 27);
+
+    ii.resize(9);
+    ii[0].set_iteration_index_and_track_algorithm(0, (int)TrackBase::TrackAlgorithm::initialStep);
+    ii[0].set_num_regions_layers(5, 72);
+
+    createPhase1TrackerGeometryAutoGen(ti, ii);
+
+    setupSteeringParamsIter0(ii[0]);
+    setupIterationParams(ii[0].m_params, 0);
+    ii[0].m_partition_seeds = PartitionSeeds0;
+    fillHitSelectionWindowsParams(ii[0]);
+
+    ii[1].Clone(ii[0]);  //added extra iterations with some preliminary setup
+    setupIterationParams(ii[1].m_params, 1);
+    ii[1].set_iteration_index_and_track_algorithm(1, (int)TrackBase::TrackAlgorithm::highPtTripletStep);
+    ii[1].set_seed_cleaning_params(2.0, 0.018, 0.018, 0.018, 0.018, 0.018, 0.05, 0.018, 0.05);
```

Comment:
also these lists of magic numbers would benefit from a brief comment ..

Reply:
Agreed, will go into README.md

Reply:
This specific line was also removed as part of removing the hardcoded parameters that are also present in the JSON files.